erpnext/regional/report/ksa_vat/ksa_vat.py

```diff
@@ -52,124 +51,74 @@ def get_data(filters):
+	get_tax_data(data, settings.ksa_vat_purchase_accounts, filters, 'Purchase Invoice')
 
 	return data
 
-def get_tax_data_for_each_vat_setting(vat_setting, filters, doctype):
-	'''
-	(KSA, {filters}, 'Sales Invoice') => 500, 153, 10 \n
-	calculates and returns \n
-	total_taxable_amount, total_taxable_adjustment_amount, total_tax'''
-	from_date = filters.get('from_date')
-	to_date = filters.get('to_date')
-
+def get_tax_data(data, settings, filters, doctype):
 	# Initiate variables
-	total_taxable_amount = 0
-	total_taxable_adjustment_amount = 0
-	total_tax = 0
+	tax_details = {}
+	for d in settings:
+		tax_details[d.account] = {
+			"title": d.title,
+			"taxable_amount": 0.00,
+			"adjustment_amount": 0.00,
+			"total_tax": 0.00
+		}
+
 	# Fetch All Invoices
-	invoices = frappe.get_all(doctype,
-	filters ={
+	invoices = frappe.get_all(doctype, filters ={
+		'company': filters.get('company'),
 		'docstatus': 1,
-		'posting_date': ['between', [from_date, to_date]]
-	}, fields =['name', 'is_return'])
-
-	for invoice in invoices:
-		invoice_items = frappe.get_all(f'{doctype} Item',
-		filters ={
-			'docstatus': 1,
-			'parent': invoice.name,
-			'item_tax_template': vat_setting.item_tax_template
-		}, fields =['item_code', 'net_amount'])
+		'posting_date': ['between', [filters.get('from_date'), filters.get('to_date')]]
+	})
 
-		for item in invoice_items:
-			# Summing up total taxable amount
-			if invoice.is_return == 0:
-				total_taxable_amount += item.net_amount
-
-			if invoice.is_return == 1:
-				total_taxable_adjustment_amount += item.net_amount
+	for inv in invoices:
+		invoice = frappe.get_doc(doctype, inv)
+		if not invoice.taxes:
+			continue
+		itemised_tax = get_itemised_tax(invoice.taxes, True)
+		itemised_taxable_amount = get_itemised_taxable_amount(invoice.items)
 
+		for item in itemised_taxable_amount.keys():
 			# Summing up total tax
-			total_tax += get_tax_amount(item.item_code, vat_setting.account, doctype, invoice.name)
-
-	return total_taxable_amount, total_taxable_adjustment_amount, total_tax
-
-
-
-def append_data(data, title, amount, adjustment_amount, vat_amount):
-	"""Returns data with appended value."""
-	data.append({"title": _(title), "amount": amount, "adjustment_amount": adjustment_amount, "vat_amount": vat_amount})
-
-def get_tax_amount(item_code, account_head, doctype, parent):
-	if doctype == 'Sales Invoice':
-		tax_doctype = 'Sales Taxes and Charges'
-
-	elif doctype == 'Purchase Invoice':
-		tax_doctype = 'Purchase Taxes and Charges'
-
-	item_wise_tax_detail = frappe.get_value(tax_doctype, {
-		'docstatus': 1,
-		'parent': parent,
-		'account_head': account_head
-	}, 'item_wise_tax_detail')
-
-	tax_amount = 0
-	if item_wise_tax_detail and len(item_wise_tax_detail) > 0:
-		item_wise_tax_detail = json.loads(item_wise_tax_detail)
-		for key, value in item_wise_tax_detail.items():
-			if key == item_code:
-				tax_amount = value[1]
-				break
-
-	return tax_amount
+			item_itemised_tax = itemised_tax.get(item)
```

Comment:
The main reason for adding the dependency of item tax template was distinguishing between zero-rated and exempted items as those have to be reported separately. That identification was done based on item tax template. How that is done in this implementation?

Reply:
Regional setup already adds `is_zero_rated` and `is_exempt` to Item doctype. Utilizing that would be better that relying on Item Tax Template. Will make changes in the code accordingly.